File: game-server/src/main/java/org/forfun/mmorpg/game/base/MessageUtil.java
package org.forfun.mmorpg.game.base;


import jforgame.socket.share.IdSession;
import jforgame.socket.share.message.Message;

import java.util.Collection;


public class MessageUtil {

	public static void pushMessage(long playerId, Message message) {
		IdSession session = GameContext.getSessionManager().getSessionBy(playerId);
		pushMessage(session, message);
	}

	public static void pushMessage(Collection<Long> playerIds, Message message) {
		for (long playerId : playerIds) {
			pushMessage(playerId, message);
		}
	}

	public static void pushMessage(IdSession session, Message message) {
		if (session == null || message == null) {
			return;
		}
		session.send(message);
	}


	public static void pushMessage(IdSession session, byte[] message) {
		if (session == null || message == null) {
			return;
		}
		session.send(message);
	}

}

File: game-server/src/main/java/org/forfun/mmorpg/game/base/MessageUtil.kt
package org.forfun.mmorpg.game.base

import jforgame.socket.share.IdSession
import jforgame.socket.share.message.Message

object MessageUtil {

    @JvmStatic
    fun pushMessage(playerId: Long, message: Message) {
        val session = GameContext.getSessionManager()?.getSessionBy(playerId)
        pushMessage(session, message)
    }

    @JvmStatic
    fun pushMessage(playerIds: Collection<Long>, message: Message) {
        for (playerId in playerIds) {
            pushMessage(playerId, message)
        }
    }

    @JvmStatic
    fun pushMessage(session: IdSession?, message: Message) {
        if (session == null || message == null) {
            return
        }
        session.send(message)
    }

    @JvmStatic
    fun pushMessage(session: IdSession?, message: ByteArray) {
        if (session == null || message == null) {
            return
        }
        session.send(message)
    }
}
